python_modules/dagster/dagster/_core/definitions/partitions/definition/time_window.py

```diff
@@ -612,22 +602,52 @@ def get_first_partition_window(self) -> Optional[TimeWindow]:
 
     @functools.lru_cache(maxsize=256)
     def _get_last_partition_window(self, current_timestamp: float) -> Optional[TimeWindow]:
```

Comment:
this is gnarly, but pretty symmetric to the implementation in _get_first_partition_window at laest